pyiceberg/table/__init__.py

```diff
@@ -4308,7 +4308,9 @@ def history(self) -> "pa.Table":
 
         return pa.Table.from_pylist(history, schema=history_schema)
 
-    def files(self, snapshot_id: Optional[int] = None) -> "pa.Table":
+    def _files(
+        self, snapshot_id: Optional[int] = None, data_file_content_types: Optional[List[DataFileContent]] = None
```

Comment:
nit: `data_file_filter` and also use Set instead of List

Reply:
What would be benefit of using a Set instead of List? `_files` is not a public function and the list also contains 2 elements at max. Are you suggesting it because class member functions are not really private, thus to prevent abuse of `_files`?

Reply:
prefer Set since it doesnt make sense to pass in multiple of the same data file types.


https://github.com/apache/iceberg-python/pull/1066/files#diff-23e8153e0fd497a9212215bd2067068f3b56fa071770c7ef326db3d3d03cee9bR4418-R4425

Reply:
Method signature of `_files` updated.